vcd/resource_vcd_vapp_network.go

```diff
@@ -219,43 +270,169 @@ func resourceVappNetworkRead(d *schema.ResourceData, meta interface{}) error {
+				newValues := map[string]interface{}{
+					"start_address": ipRange.StartAddress,
+					"end_address":   ipRange.EndAddress,
+				}
+				staticIpRanges.Add(newValues)
+			}
+			err = d.Set("static_ip_pool", staticIpRanges)
+			if err != nil {
+				return fmt.Errorf("[vApp network static pool read] set issue: %s", err)
+			}
+		}
+
 		// TODO adjust when we have option to switch between API versions or upgrade the default version
 		// API does not return GuestVlanAllowed if API client version is 27.0 (default at the moment) therefore we rely
 		// on updating statefile only if the field was returned. In API v31.0 - the field is returned.
-		if c.GuestVlanAllowed != nil {
-			err = d.Set("guest_vlan_allowed", *c.GuestVlanAllowed)
+		if config.GuestVlanAllowed != nil {
+			err = d.Set("guest_vlan_allowed", *config.GuestVlanAllowed)
 			if err != nil {
 				return err
 			}
 		}
+		if config.ParentNetwork != nil {
+			_ = d.Set("org_network", config.ParentNetwork.Name)
+		}
+		if config.Features != nil && config.Features.FirewallService != nil {
+			_ = d.Set("firewall_enabled", vAppNetwork.Configuration.Features.FirewallService.IsEnabled)
+		}
+		if config.Features != nil && config.Features.NatService != nil {
+			_ = d.Set("nat_enabled", config.Features.NatService.IsEnabled)
+		}
+		_ = d.Set("retain_ip_mac_enabled", config.RetainNetInfoAcrossDeployments)
 	}
-
 	return nil
 }
 
+func resourceVappNetworkUpdate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentVapp(d)
+	defer vcdClient.unLockParentVapp(d)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	vapp, err := vdc.GetVAppByName(d.Get("vapp_name").(string), false)
+	if err != nil {
+		return fmt.Errorf("error finding vApp. %#v", err)
+	}
+
+	staticIpRanges, err := expandIPRange(d.Get("static_ip_pool").(*schema.Set).List())
+	if err != nil {
+		return err
+	}
+
+	natEnabled := d.Get("nat_enabled").(bool)
+	fwEnabled := d.Get("firewall_enabled").(bool)
+	retainIpMacEnabled := d.Get("retain_ip_mac_enabled").(bool)
+
+	vappNetworkSettings := &govcd.VappNetworkSettings{
+		ID:                 d.Id(),
+		Name:               d.Get("name").(string),
+		Description:        d.Get("description").(string),
+		Gateway:            d.Get("gateway").(string),
+		NetMask:            d.Get("netmask").(string),
+		DNS1:               d.Get("dns1").(string),
+		DNS2:               d.Get("dns2").(string),
+		DNSSuffix:          d.Get("dns_suffix").(string),
+		StaticIPRanges:     staticIpRanges.IPRange,
+		NatEnabled:         &natEnabled,
+		FirewallEnabled:    &fwEnabled,
+		RetainIpMacEnabled: &retainIpMacEnabled,
+	}
+
+	if _, ok := d.GetOk("guest_vlan_allowed"); ok {
+		convertedValue := d.Get("guest_vlan_allowed").(bool)
+		vappNetworkSettings.GuestVLANAllowed = &convertedValue
+	}
+
+	if dhcp, ok := d.GetOk("dhcp_pool"); ok && len(dhcp.(*schema.Set).List()) > 0 {
```

Comment:
Looks like at least this block could be shared between `create` and `update`

Reply:
done